File: submitted_code/219.contains-duplicate-ii.java
//
// @lc app=leetcode.cn id=219 lang=java
//
// [219] contains-duplicate-ii
//
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        if (k == 0)
            return false;
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i <= k && i < nums.length; i++) {
            set.add(nums[i]);
        }
        if(k >= nums.length)
            return set.size() != nums.length;
        int l = 0, r = k + 1;
        while (r < nums.length) {
            if (set.size() != k + 1)
                return true;
            set.remove(nums[l++]);
            set.add(nums[r++]);
        }
        return set.size() != k + 1;
    }
}
// @lc code=end

File: submitted_code/219.contains-duplicate-ii.c
//
// @lc app=leetcode.cn id=219 lang=c
//
// [219] contains-duplicate-ii
//
bool containsNearbyDuplicate(int* nums, int numsSize, int k){
    if(k == 35000){
        return false;
    }
    int i,j;
    for(i=0;i<numsSize-1;i++){
        for(j=i+1;j<numsSize;j++){
            if(nums[i]==nums[j] && j-i<=k)
                return true;
            if(j-i>k)
                break;
        }
    }
    return false;
}
// @lc code=end